test/e2e/framework/addon/vault/setup.go

```diff
@@ -146,39 +152,55 @@ func (v *VaultInitializer) Init() error {
 	return nil
 }
 
+// Set up a Vault PKI.
 func (v *VaultInitializer) Setup() error {
+	// Enable a new Vault secrets engine at v.RootMount
 	if err := v.mountPKI(v.RootMount, "87600h"); err != nil {
 		return err
 	}
 
+	// Generate a self-signed CA cert using the engine at v.RootMount
 	rootCa, err := v.generateRootCert()
```

Comment:
suggestion (non-blocking): it's 100% not required because I think it's beyond the scope of this PR, but it would be good to change `generateRootCert` and `generateIntermediateSigningReq` to generate ECDSA certs rather than the RSA cert they currently default to. that'll improve the speed of the tests and reduce the load on the Vault server during the tests.